src/server/twemcache/data/process.c

```diff
@@ -242,23 +245,25 @@ _process_set(struct response *rsp, struct request *req)
     struct item *it;
     struct bstring key;
 
-    status = _put(&istatus, req);
+    status = _put(&istatus, req, true);
```

Comment:
why are you deleting the existing item if `set` fails? If I understand this correctly, failing to set `key` to `value2` will delete existing `value1`, is that so?